docs/dev/intro-v3-engine.md

```diff
@@ -0,0 +1,104 @@
+
+### Why Apache Calcite?
+
+Introducing Apache Calcite brings serval significant advantages:
+
+1. Enhanced query plan optimization capabilities: Calcite's optimizer can effectively optimize execution plans for both complex SQL and PPL queries.
+
+2. Simplified development of new commands and functions: Expanding PPL commands is one of the key targets to enhancing the PPL language. Calcite helps streamline the development cycle for new commands and functions.
+
+3. Decoupled execution layer: Calcite can be used for both query optimization and execution, or solely for query optimization while delegating execution to other backends such as DataFusion or Velox."
+
+Find more details in [V3 Architecture](./intro-v3-architecture.md).
+
+---
+## 2. What's New
+
+In the initial release of the V3 engine (3.0.0-beta), the main new features focus on enhancing the PPL language while maintaining maximum compatibility with V2 behavior.
+
+* **[Join](../user/ppl/cmd/join.rst) Command**
+* **[Lookup](../user/ppl/cmd/lookup.rst) Command**
+* **[Subquery](../user/ppl/cmd/subquery.rst) Command**
+
+---
+## 3.What are Changed
+
+### 3.1 Breaking Changes
+
+Because of implementation changed internally, following behaviors are changed from 3.0.0-beta. (Behaviors in V3 is correct)
+
+|                       Item                       |    V2     |          V3          |
+|:------------------------------------------------:|:---------:|:--------------------:|
+|          Return type of `timestampdiff`          | timestamp |         int          |
+|             Return type of `regexp`              |    int    |       boolean        |
+|   Return type of `count`,`dc`,`distinct_count`   |    int    |        bigint        |
+|     Return type of `ceiling`,`floor`,`sign`      |    int    | same type with input |
+| like(firstname, 'Ambe_') on value "Amber JOHnny" |   true    |        false         |
+| like(firstname, 'Ambe*') on value "Amber JOHnny" |   true    |        false         |
+|            cast(firstname as boolean)            |   false   |         null         |
+| Sum multiple `null` values when pushdown enabled |     0     |         null         |
+
+
+### 3.2 Fallback Mechanism
+
+As v3 engine is experimental in 3.0.0-beta, not all PPL commands could work under this new engine. Those unsupported queries will be forwarded to V2 engine by fallback mechanism. To avoid impact on your side, normally you won't see any difference in a query response. If you want to check if and why your query falls back to be handled by V2 engine, please check OpenSearch log for "Fallback to V2 query engine since ...".
+
+### 3.3 Limitations
+
+For the following functionalities in V3 engine, the query will be forwarded to the V2 query engine and thus you cannot use new features in [2. What's New](#2-whats-new).
+
+#### Unsupported functionalities
+- All SQL queries
+- `trendline`
+- `show datasource`
+- `explain`
```

Comment:
So we cannot check the query plan supported by V3 engine?

Reply:
Yes. The original code freeze is March 31. So it is out of P0 tasks for beta. But seems the deadline postponed to April 8, I will add this command if possible.

Reply:
Discussed offline with Charlie, it requires a RFC for details of new explain command.